homeassistant/components/fan/wemo.py

```diff
@@ -111,29 +117,48 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     def service_handle(service):
         """Handle the WeMo humidifier services."""
         entity_ids = service.data.get(ATTR_ENTITY_ID)
-        target_humidity = service.data.get(ATTR_TARGET_HUMIDITY)
-
-        if entity_ids:
-            humidifiers = [device for device in hass.data[DATA_KEY].values() if
-                           device.entity_id in entity_ids]
-        else:
-            humidifiers = hass.data[DATA_KEY].values()
 
-        for humidifier in humidifiers:
-            humidifier.set_humidity(target_humidity)
+        if service.service == SERVICE_SET_HUMIDITY:
+            target_humidity = service.data.get(ATTR_TARGET_HUMIDITY)
+
+            if entity_ids:
```

Comment:
This block is the same for both services. Do it before selecting service to call.